### ISSUE TITLE
PNNStrategy model top_1_acc is poor in older test_stream experiences.

### ISSUE THREAD

**Issue by Aritro-94** [ContinualAI/avalanche#1100]
`benchmark = SplitMNIST(n_experiences=2, shuffle=True,return_task_id=False)
`
`model = PNN()
``cl_strategy = PNNStrategy()
`
After training on both the experiences in train_stream using the continuous learning strategy, the model has good Top_1_Acc in the corresponding latest test_stream experience, but has almost 0 accuracy in the previous test_stream experience

The result of `cl_strategy.eval(benchmark.test_stream)` has been pasted below

![eval](https://user-images.githubusercontent.com/52375353/181362969-9125f605-8e30-4ac6-9de0-62007c6d67ce.png)

However, it is working perfectly fine when return_task_id is set to True and the model.forward receives an appropriate task_id istead of just 0.

For your reference I am pasting the code block below:

```
import torch
from avalanche.models import PNN
from torch.nn import CrossEntropyLoss
from torch.optim import SGD
from avalanche.benchmarks.classic import SplitMNIST
from avalanche.training.plugins import EvaluationPlugin
from avalanche.evaluation.metrics import accuracy_metrics
from avalanche.training.supervised import PNNStrategy

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

benchmark = SplitMNIST(n_experiences=2, shuffle=True,return_task_id=False)

model = PNN()

optimizer = SGD(model.parameters(), lr=0.001, momentum=0.9)

criterion = CrossEntropyLoss()
eval_plugin = EvaluationPlugin(
    accuracy_metrics(epoch=True,
                     experience=True))
cl_strategy = PNNStrategy(
    model, optimizer, criterion, train_mb_size=16, train_epochs=3,
    eval_mb_size=32, evaluator=eval_plugin, device=device)


results = []
for task_id,train_task in enumerate(benchmark.train_stream):
    cl_strategy.train(train_task, num_workers=4)
    results.append(cl_strategy.eval(benchmark.test_stream))

cl_strategy.eval(benchmark.test_stream)
```

Kindly help.

**Comment by AntonioCarta** [comment]
This is correct. PNN need task labels. Without them, the model does not work.

**Comment by Aritro-94** [comment]
In production the task label will not be obtainable. Could you suggest which model to use which learns incrementally without task_id.

**Comment by AntonioCarta** [comment]
Replay is the safest choice.
